lib/ts/recipe/emailpassword/utils.ts

```diff
@@ -343,6 +343,9 @@ export function getFormattedFormField(field: NormalisedFormField): NormalisedFor
         validate: async (value: any): Promise<string | undefined> => {
             // Absent or not optional empty field
             if (value === "" && field.optional === false) {
+                if (field.nonOptionalErrorMsg !== undefined && field.nonOptionalErrorMsg !== "") {
```

Comment:
why are you checking for `!== ""`?

Reply:
the user will get the standard usual error Field is not optional
if the value of nonOptionalErrorMsg is set to empty string